2016-02-11 12:12:55,892 5667 ERROR odoo-test openerp.tools.safe_eval: Cannot eval u'a+b+c'
Traceback (most recent call last):
  File "/home/odoo-test/instance/release/parts/openerp/openerp/tools/safe_eval.py", line 288, in safe_eval
    return eval(test_expr(expr, _SAFE_OPCODES, mode=mode), globals_dict, locals_dict)
  File "", line 1, in <module>
NameError: name 'a' is not defined
2016-02-11 12:12:55,894 5667 ERROR odoo-test openerp.tools.safe_eval: Cannot eval u'd+e'
Traceback (most recent call last):
  File "/home/odoo-test/instance/release/parts/openerp/openerp/tools/safe_eval.py", line 288, in safe_eval
    return eval(test_expr(expr, _SAFE_OPCODES, mode=mode), globals_dict, locals_dict)
  File "", line 1, in <module>
NameError: name 'd' is not defined
2016-02-11 12:12:55,895 5667 ERROR odoo-test openerp.tools.safe_eval: Cannot eval u'a+b+c'
Traceback (most recent call last):
  File "/home/odoo-test/instance/release/parts/openerp/openerp/tools/safe_eval.py", line 288, in safe_eval
    return eval(test_expr(expr, _SAFE_OPCODES, mode=mode), globals_dict, locals_dict)
  File "", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'

![image](https://cloud.githubusercontent.com/assets/7602419/12976036/668519b8-d0c1-11e5-8dd4-d48086669f1a.png)

--- a/mis_builder/models/mis_builder.py
+++ b/mis_builder/models/mis_builder.py
@@ -651,6 +651,7 @@ class MisReportInstancePeriod(orm.Model):
                     kpi_val_comment = kpi.name + " = " + kpi.expression
                     kpi_eval_expression = aep.replace_expr(kpi.expression)
                     kpi_val = safe_eval(kpi_eval_expression, localdict)
+                    localdict[kpi.name] = kpi_val
                 except ZeroDivisionError:
                     kpi_val = None
                     kpi_val_rendered = '#DIV/0'
@@ -668,7 +669,6 @@ class MisReportInstancePeriod(orm.Model):
                     kpi_val_rendered = kpi_obj.render(
                         cr, uid, lang_id, kpi, kpi_val, context=context)
 
-                localdict[kpi.name] = kpi_val
                 try:
                     kpi_style = None
                     if kpi.css_style:

